app/controllers/users/registrations_controller.rb

```diff
@@ -74,4 +78,37 @@ def verify_via_recaptcha!
   def handle_recaptcha_failure
     render json: {error: "There was an temporary error preventing your payment. Please try again. If it persists, please contact support@commitchange.com with error code: 5X4J "}, status: :unprocessable_entity
   end
+
+  def update_params
+    params[:user].except(:otp_required_for_login)
+  end
+
+  def handle_two_factor_changes
+    otp_param = params[:user][:otp_required_for_login]
+    return unless %w[true false].include?(otp_param)
+
+    if otp_param == "true" && !current_user.otp_required_for_login?
+      # Enable 2FA
+      current_user.otp_required_for_login = true
+      current_user.otp_secret = User.generate_otp_secret
+      @two_factor_enabled = true
```

Comment:
Not a blocker but this feels like something that should be in the `User` as an instance variable that reset after a save.

Reply:
> Not sure I follow. Seems weird to add a dedicated variable to the User to see if `otp_required_for_login` changed since we already have that `otp_required_for_login` field on the user. Are you saying this registrations controller should use e.g. `current_user.saved_change_to_otp_required_for_login?` to determine which message to report to the user insteaad of this `@two_factor_enabled` variable? That seems reasonable to me.

That's better than my idea :laughing:

Ya, keeping track of whether a particular field changed should be handled by the `User` instead of the controller.

Reply:
Did that and was able to get rid of the instance variable too.